scripts/github-actions-time.py

```diff
@@ -0,0 +1,170 @@
+import os
+import requests
+import json
+from datetime import datetime, timedelta, timezone, date
+
+GITHUB_API_URL = "https://api.github.com"
+REPO_OWNER = "telepresenceio"
+REPO_NAME = "telepresence"
+
+def dump_to_file(data, filename):
+    with open(filename, "w") as f:
+        json.dump(data, f)
+
+def dict_from_json_file(filename):
+    with open(filename, "r") as f:
+        return json.load(f)
+
+
+def fetch_workflow_runs(token, date_threshold, days):
+    print(f"Fetching workflow runs from {date_threshold} to {days} days ago.")
+
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Accept": "application/vnd.github+json"
+    }
+    url = f"{GITHUB_API_URL}/repos/{REPO_OWNER}/{REPO_NAME}/actions/runs"
+
+    params = {
+        "status": "completed",
+        "created": f">{date_threshold}"
+    }
+
+    if os.path.exists(f"workflow_runs-{days}days-to-{date_threshold}.json"):
+        return dict_from_json_file(f"workflow_runs-{days}days-to-{date_threshold}.json")
```

Comment:
minor,  I think doing this before setting up the HTTP variables makes the intent of the function more clear:  if the JSON file exists return the data, otherwise get it from GitHub.